Xamarin.Forms.Platform.Android/AppCompat/CheckBoxRendererBase.cs

```diff
@@ -0,0 +1,272 @@
+using System;
+using System.ComponentModel;
+using Android.Content;
+using Android.Support.V7.Widget;
+using AView = Android.Views.View;
+using Android.Views;
+using Xamarin.Forms.Internals;
+using AColor = Android.Graphics.Color;
+using Android.Graphics;
+using Xamarin.Forms.Platform.Android.FastRenderers;
+using Android.Widget;
+using Android.Content.Res;
+using Android.Support.V4.Widget;
+using AAttribute = Android.Resource.Attribute;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	public class CheckBoxRendererBase :
+		AppCompatCheckBox,
+		IVisualElementRenderer,
+		AView.IOnFocusChangeListener,
+		CompoundButton.IOnCheckedChangeListener,
+		ITabStop
+	{
+		bool _disposed;
+		bool _skipInvalidate;
+		int? _defaultLabelFor;
+		VisualElementTracker _tracker;
+		VisualElementRenderer _visualElementRenderer;
+		IPlatformElementConfiguration<PlatformConfiguration.Android, CheckBox> _platformElementConfiguration;
+		CheckBox _checkBox;
+
+		static int[][] _checkedStates = new int[][]
+					{
+						new int[] { AAttribute.StateEnabled, AAttribute.StateChecked },
+						new int[] { AAttribute.StateEnabled, -AAttribute.StateChecked },
+						new int[] { -AAttribute.StateEnabled, AAttribute.StateChecked },
+						new int[] { -AAttribute.StateEnabled, -AAttribute.StatePressed },
+					};
+
+		public event EventHandler<VisualElementChangedEventArgs> ElementChanged;
+		public event EventHandler<PropertyChangedEventArgs> ElementPropertyChanged;
+
+		public CheckBoxRendererBase(Context context) : base(context) => Init();
+
+		public CheckBoxRendererBase(Context context, int defStyleAttr) : base(context, null, defStyleAttr) => Init();
+
+		void Init()
+		{
+			SoundEffectsEnabled = false;
+			SetOnCheckedChangeListener(this);
+			Tag = this;
+			OnFocusChangeListener = this;
+			ClipToOutline = true;
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (disposing && !_disposed)
```

Comment:
If `_disposed` is true, we don't need to call `base.Dispose(disposing)` - it's already been called. We can just return immediately.